profiler/profiler.go

```diff
@@ -249,33 +209,6 @@ func (p *profiler) run() {
 		_, ok := p.cfg.types[t]
 		return ok
 	}
-	p.telemetry.Start(
-		// Integrations are part of this API to match the telemetry API,
-		// but there aren't really "integrations" for the profiler.
-		// Note that if we did have integrations, we'd need a mechanism
-		// to report the individual packages. runtime/debug.BuildInfo
-		// only shows *modules* that we depend on, not individual
-		// packages.
-		[]telemetry.Integration{},
-		[]telemetry.Configuration{
-			{Name: "delta_profiles", Value: p.cfg.deltaProfiles},
-			{Name: "agentless", Value: p.cfg.agentless},
-			{Name: "profile_period", Value: p.cfg.period.String()},
-			{Name: "cpu_duration", Value: p.cfg.cpuDuration.String()},
-			{Name: "cpu_profile_rate", Value: p.cfg.cpuProfileRate},
-			{Name: "block_profile_rate", Value: p.cfg.blockRate},
-			{Name: "mutex_profile_fraction", Value: p.cfg.mutexFraction},
-			{Name: "max_goroutines_wait", Value: p.cfg.maxGoroutinesWait},
-			{Name: "cpu_profile_enabled", Value: profileEnabled(CPUProfile)},
-			{Name: "heap_profile_enabled", Value: profileEnabled(HeapProfile)},
-			{Name: "block_profile_enabled", Value: profileEnabled(BlockProfile)},
-			{Name: "mutex_profile_enabled", Value: profileEnabled(MutexProfile)},
-			{Name: "goroutine_profile_enabled", Value: profileEnabled(GoroutineProfile)},
-			{Name: "goroutine_wait_profile_enabled", Value: profileEnabled(expGoroutineWaitProfile)},
-			{Name: "upload_timeout", Value: p.cfg.uploadTimeout.String()},
-		},
```

Comment:
Are these just not part of v2? I'm trying to understand why all of this telemetry data is being pulled out of the profiler.

Reply:
I decided to remove telemetry completely from the profiler because of the expectation for the profiler and tracer to use a single `GlobalClient` for the future. Under this v2 implementation the tracer would be responsible for starting the telemetry client and the profiler would use `ProductEnabled` to signal that it has started. 

This PR https://github.com/DataDog/dd-trace-go/pull/1769 would be responsible for actually adding telemetry info to the profiler and tracer. I believe the profiling team is fine with telemetry being temporarily removed? @nsrip-dd  https://github.com/DataDog/dd-trace-go/pull/1807#issuecomment-1476202054

Reply:
Since the changes in the client invalidate the profiler's current usage, I'm fine with the usage going away and getting re-implemented in a subsequent PR.